docs/source/platform/federation.mdx

```diff
@@ -0,0 +1,228 @@
+
+## Registering your implementing services
+
+If you're running a distributed GraphQL infrastructure, where implementing services [compose](https://www.apollographql.com/docs/apollo-server/federation/federation-spec/#fetch-service-capabilities) to form a complete schema, there's functionality to register each service's partial schema and still get all of the benefits of the schema registry.
+
+To push a single implementing service to Apollo, run `apollo service:push` with the `--serviceName` and `--endpoint` tags. The CLI will know where to fetch your local partial schema based on this endpoint. Note that running `apollo service:push` on any implementing service also registers the overall schema of the graph and writes files to GCS.
+
+As an example, running service push on a local “Accounts” implementing service might look like:
+
+```
+~$apollo service:push --serviceName="Accounts" --serviceURL="http://accounts-graphql:4001/ --endpoint="http://localhost:4001/"
+  ✔ Loading Apollo Project
+  ✔ Uploading service to Engine
+id      schema                        tag
+──────  ────────────────────────────  ───────
+190331  service-list-federation-demo  current
+```
+
+Pushing an implementing service to Apollo will update the [Services tab](#inspecting-your-graph).
+
+## Inspecting your Graph
+> Using the `apollo service:list` command or the UI to understand your graph
+
+To understand the implementing services that make up your graph, you can use both the Apollo command line interface as well as the Apollo Platform.
+
+Use `apollo service:list` in the command line to see a snapshot of the services that make up your graph, including their endpoints and when they were last updated.
+
+Here's what running `apollo service:list` will look like:
+
+```
+~$ apollo service:list
+  ✔ Loading Apollo Project
+  ✔ Fetching list of services for graph service-list-federation-demo
+
+name       URL                            last updated
+─────────  ─────────────────────────────  ────────────────────────
+Accounts   http://localhost:4001/graphql  3 July 2019 (2 days ago)
+Inventory  http://localhost:4004/graphql  3 July 2019 (2 days ago)
+Products   http://localhost:4003/graphql  3 July 2019 (2 days ago)
+Reviews    http://localhost:4002/graphql  3 July 2019 (2 days ago)
+
+View full details at: https://engine.apollographql.com/graph/service-list-federation-demo/service-list
+```
+
+Navigate to the Services tab in the Apollo Platform to see an interactive list of information about the implementing services in your graph.
+
+<div style="text-align:center">
+    <img src="../images/ServicesListPlatform.png" alt="Service List">
+    </img>
+</div>
+
+[//]: # (Description: How to see the current services in your graph. This section might be expanded later to discuss showing history, composition results, etc. This section should include some screenshots)
+[//]: # (Assignee: Maya)
+[//]: # (Reviewer: Justin)
+
+## Validating changes to your graph
```

Comment:
Pasting some examples here so that we can use them in the section.

Reply:
EXAMPLE of what `apollo service:check` looks like with SAFE changes:
```
$ npx apollo@2.15.1-alpha.1 service:check --serviceName registry --endpoint http://localhost:4001/api/graphql --engine https://engine-staging-graphql.apollographql.com/api/graphql --tag staging --frontend https://engine-staging.apollographql.com
  ✔ Loading Apollo Project
  ✔ Found 0 graph composition errors for service registry on graph apollo-federated
  ✔ Validated composed schema against tag staging on graph apollo-federated
  ✔ Compared 2 schema changes against 56 operations over the last 548 days
  ✔ Found 0 breaking changes and 2 compatible changes

PASS    FIELD_ADDED  `ServiceMutation.createApiKeyAndWriteStorageSecret` was added
PASS    FIELD_ADDED  `ServiceMutation.uploadSchema` was added

View full details at: https://engine-staging.apollographql.com/service/apollo-federated/check/5f209250-000e-45f4-9d5c-ff022225f542
```

Reply:
EXAMPLE of what `apollo service:check` looks like with UNSAFE changes:
```
$ ../apollo-tooling/packages/apollo/bin/run service:check --serviceName registry --endpoint http://localhost:4001/api/graphql --engine https://engine-staging-graphql.apollographql.com/api/graphql --tag staging --frontend https://engine-staging.apollographql.com
  ✔ Loading Apollo Project
  ✔ Found 0 graph composition errors for service registry on graph apollo-federated
  ✔ Validated composed schema against tag staging on graph apollo-federated
  ✔ Compared 1 schema change against 55 operations over the last 548 days
  ✖ Found 1 breaking change and 0 compatible changes
    → breaking changes found

FAIL    FIELD_REMOVED  `ServiceMutation.uploadSchema` was removed

View full details at: https://engine-staging.apollographql.com/service/apollo-federated/check/941c4fa2-c716-4df4-9fc5-130b88c8ebca
```

Reply:
Add a note that validation does not update the graph

Reply:
EXAMPLE of what `apollo service:check` looks like with composition errors:
```
$ ../apollo-tooling/packages/apollo/bin/run service:check --serviceName registry --endpoint http://localhost:4001/api/graphql --engine https://engine-staging-graphql.apollographql.com/api/graphql --tag staging
  ✔ Loading Apollo Project
  ✖ Found 16 graph composition errors for service registry on graph apollo-federated
    → Federated service composition was unsuccessful. Please see the reasons below.
    Validating composed schema against tag staging on graph apollo-federated
    Comparing schema changes
    Reporting result

Service   Field                                                                   Message
────────  ──────────────────────────────────────────────────────────────────────  ───────────────────────────────────────────────────────
registry  ServiceMutation.upsertImplementingService                               requires the field `service` to be marked as @external.
registry  ServiceMutation.upsertImplementingServiceAndTriggerComposition          requires the field `service` to be marked as @external.
registry  ServiceMutation.uploadPartialSchema                                     requires the field `service` to be marked as @external.
registry  ServiceMutation.validatePartialSchemaOfImplementingServiceAgainstGraph  requires the field `service` to be marked as @external.
registry  ServiceMutation.recalculateCompositionConfig                            requires the field `service` to be marked as @external.
registry  ServiceMutation.removeImplementingServiceAndTriggerComposition          requires the field `service` to be marked as @external.
registry  ServiceMutation.removeImplementingService                               requires the field `service` to be marked as @external.
registry  ServiceMutation.checkPartialSchema                                      requires the field `service` to be marked as @external.
registry  ServiceMutation.registerOperations                                      requires the field `service` to be marked as @external.
registry  ServiceMutation.registerOperationsWithResponse                          requires the field `service` to be marked as @external.
registry  ServiceMutation.uploadSchema                                            requires the field `service` to be marked as @external.
registry  ServiceMutation.deleteSchema                                            requires the field `service` to be marked as @external.
registry  ServiceMutation.deleteInvalidSchema                                     requires the field `service` to be marked as @external.
registry  ServiceMutation.checkSchema                                             requires the field `service` to be marked as @external.
registry  ServiceMutation.tagSchema                                               requires the field `service` to be marked as @external.
registry  ServiceMutation.deleteSchemaTag                                         requires the field `service` to be marked as @external.
```
We should obviously pare these composition errors down.